### PR TITLE
Display paths to config/urls/cache in --help

### LINKED ISSUE #294: Whereis the default config file?
newsboat 2.10.2 - https://newsboat.org/
Copyright (C) 2006-2015 Andreas Krennmair
Copyright (C) 2015-2017 Alexander Batischev
Copyright (C) 2006-2017 Newsbeuter contributors
Copyright (C) 2017 Newsboat contributors

newsboat is free software and licensed under the MIT License.
Tapez `newsboat -vv' pour obtenir des informations supplémentaires.

newsboat 2.10.2
System: Linux 4.14.65 (x86_64)
Compiler: g++ 7.3.0
ncurses: ncurses 6.1.20180127 (compiled with 6.1)
libcurl: libcurl/7.61.1 OpenSSL/1.0.2p zlib/1.2.11 (compiled with 7.61.1)
SQLite: 3.24.0 (compiled with 3.24.0)
libxml2: compiled with 2.9.8

config file is not created by newsboat.... so I searched  where to write the (well documented) options.
thing is, neither
https://github.com/newsboat/newsboat/blob/master/doc/newsboat.txt
nor
newsboat -h
is any help here: you have to guess that it is in $HOME/newsboat/config
It sure is in the man page, but then again, not in the "-C" section...
Please update man page, website and "-h" quick help.

### PR DESCRIPTION
**Pull request by Minoru**
Fixes #294.

I'll merge this in three days unless someone—anyone—stops me for a review.

### CODE REVIEW

File: newsboat.cpp [hunk 1]
```diff
@@ -188,7 +207,22 @@ int main(int argc, char* argv[])
 	CliArgsParser args(argc, argv);
 
 	if (args.should_print_usage()) {
-		print_usage(args.program_name());
+		std::string config_path = utils::resolve_tilde("~/.newsboat/config");
+		if (args.set_config_file()) {
+			config_path = args.config_file();
+		}
```
Comment (dennisschagt) [inline newsboat.cpp]: I expect this does not result in the actual location after creating files in XGD locations (`~/.config/newsboat/xxx`) which might be confusing.

Same for `urls_path` and `cache_path`.

Comment (Minoru) [reply newsboat.cpp]: Wow, that's embarrassing. My code actually makes no sense at all!

Thank you for the great catch :) I think I should move `ConfigPaths` out of the `Controller` and into `main()` to fix this; let me try that.

Comment (Minoru) [reply newsboat.cpp]: Here's proof that with new commit, everything works as intended. I redacted the options to make it a bit shorter.

```
# I don't have XDG directories, so Newsboat uses dotdir
$ XDG_CONFIG_HOME= XDG_DATA_HOME= ./newsboat --help
Files:
	- config:  /home/minoru/.newsboat/config
	- urls:    /home/minoru/.newsboat/urls
	- cache:   /home/minoru/.newsboat/cache.db

$ HOME=/wonderland XDG_CONFIG_HOME= XDG_DATA_HOME= ./newsboat --help
Files:
	- config:  /wonderland/.newsboat/config
	- urls:    /wonderland/.newsboat/urls
	- cache:   /wonderland/.newsboat/cache.db

# /dev/shm is tmpfs
$ mkdir /dev/shm/newsboat-xdg-test

# For the following tests, note that Newsboat switches to XDG *only* if XDG
# config directory for Newsboat exist

# Neither config nor data dirs for Newsboat exist, so falling back to dotdir
$ HOME=/home/minoru XDG_CONFIG_HOME= XDG_DATA_HOME=/dev/shm/newsboat-xdg-test ./newsboat --help
Files:
	- config:  /home/minoru/.newsboat/config
	- urls:    /home/minoru/.newsboat/urls
	- cache:   /home/minoru/.newsboat/cache.db

# Still neither directory exists, so dotdir again
$ HOME=/home/minoru XDG_CONFIG_HOME=/dev/shm/newsboat-xdg-test XDG_DATA_HOME=/dev/shm/newsboat-xdg-test ./newsboat --help
Files:
	- config:  /dev/shm/newsboat-xdg-test/newsboat/config
	- urls:    /dev/shm/newsboat-xdg-test/newsboat/urls
	- cache:   /dev/shm/newsboat-xdg-test/newsboat/cache.db

# Ditto
$ HOME=/home/minoru XDG_CONFIG_HOME=/dev/shm/newsboat-xdg-test XDG_DATA_HOME= ./newsboat --help
Files:
	- config:  /dev/shm/newsboat-xdg-test/newsboat/config
	- urls:    /dev/shm/newsboat-xdg-test/newsboat/urls
	- cache:   /home/minoru/.local/share/newsboat/cache.db

# Now we have a directory for Newsboat
$ mkdir /dev/shm/newsboat-xdg-test/newsboat

# XDG_CONFIG_HOME is not set, so defaults to ~/.config. The ~/.config/newsboat
# doesn't exist, so falling back to dotdir
$ HOME=/home/minoru XDG_CONFIG_HOME= XDG_DATA_HOME=/dev/shm/newsboat-xdg-test ./newsboat --help
Files:
	- config:  /home/minoru/.newsboat/config
	- urls:    /home/minoru/.newsboat/urls
	- cache:   /home/minoru/.newsboat/cache.db

# Finally found the config dir for Newsboat, so using XDG
$ HOME=/home/minoru XDG_CONFIG_HOME=/dev/shm/newsboat-xdg-test XDG_DATA_HOME=/dev/shm/newsboat-xdg-test ./newsboat --help
Files:
	- config:  /dev/shm/newsboat-xdg-test/newsboat/config
	- urls:    /dev/shm/newsboat-xdg-test/newsboat/urls
	- cache:   /dev/shm/newsboat-xdg-test/newsboat/cache.db

# Config dir for Newsboat found, so using XDG -- but XDG_DATA_HOME is not set, so default
$ HOME=/home/minoru XDG_CONFIG_HOME=/dev/shm/newsboat-xdg-test XDG_DATA_HOME= ./newsboat --help
Files:
	- config:  /dev/shm/newsboat-xdg-test/newsboat/config
	- urls:    /dev/shm/newsboat-xdg-test/newsboat/urls
	- cache:   /home/minoru/.local/share/newsboat/cache.db
```

Comment (dennisschagt) [reply newsboat.cpp]: Nice solution, looks good to me.

Just to be sure, the comment `# Still neither directory exists, so dotdir again` shows XDG directories. Am I misunderstanding something there?

Comment (Minoru) [reply newsboat.cpp]: My bad: I originally run the commands without redefining `HOME`, but then decided to do it in all commands for clarity. So I re-run the commands, but forgot to delete /dev/shm/newsboat-xdg-test beforehand. The proof is now amended. Thanks for being so attentive!

**Comment by Minoru** [comment]
I guess this is done, then. Thanks for the review, @dennisschagt; I'll squash and merge this now.

**Pull request merged**